Lib/types.py

```diff
@@ -300,4 +300,81 @@ def wrapped(*args, **kwargs):
 NoneType = type(None)
 NotImplementedType = type(NotImplemented)
 
+class TracebackGroup:
+    def __init__(self, excs, frame):
+        self.tb_frame = frame
+        self.tb_next_map = {} # TODO: make it a weak key dict
+        for e in excs:
+            self.tb_next_map[e] = e.__traceback__
+
+class ExceptionGroup(BaseException):
+
+    def __init__(self, excs, frame=None):
+        self.excs = set(excs)
+        self.frame = frame or sys._getframe()
+        # self.__traceback__ is updated as usual, but self.__traceback_group__
+        # is the frame where the exception group was created (and it is
+        # preserved on splits).  So __traceback_group__ + __traceback__
+        # gives us the full path.
+        import types
+        self.__traceback__ = types.TracebackType(None, self.frame, 0, 0)
+        self.__traceback_group__ = TracebackGroup(self.excs, self.frame)
+
+    def split(self, E):
```

Comment:
Now that we have `split()`, what do you think `join()` will look like? A class method? I was also thinking perhaps it can be spelled as `+`.

Reply:
Do we need join? Or do we just make a new list of exceptions (some of which can be EGs) and construct a new EG (which takes care or creating the new TBG)?

Reply:
Not sure, but if you split and then join, the pieces the result should be the original EG no matter how the split turned out.